### ISSUE TITLE
Update TripletNetwork and Triplet Loss to TensorFlow and Keras

### ISSUE THREAD

**Issue by VecherVhatuX** [VecherVhatuX/learning_data_science_in_prison#1282]
The existing codebase uses PyTorch but needs to be updated to use TensorFlow and Keras for better compatibility and flexibility. The main issues are that the TripletNetwork class and the triplet loss function need to be rewritten using TensorFlow and Keras components.
